cmd/system-probe/modules/network_tracer.go

```diff
@@ -30,6 +30,11 @@ var inactivityLogDuration = 10 * time.Minute
 var NetworkTracer = api.Factory{
 	Name: "network_tracer",
 	Fn: func(cfg *config.AgentConfig) (api.Module, error) {
+		if !cfg.CheckIsEnabled("Network") {
```

Comment:
What about using the `api.Factory` `Name` ? We could then call `CheckIsEnabled` before entering `Fn` here : https://github.com/DataDog/datadog-agent/blob/master/cmd/system-probe/loader.go#L29

Reply:
This is mimicking how TCP Queue Length and OOM Kill do things. I believe the plan is to refactor some of this in the near future when system-probe will be made to work optionally with/without the security module.  
https://github.com/DataDog/datadog-agent/blob/master/cmd/system-probe/modules/oom_kill_probe.go#L18
https://github.com/DataDog/datadog-agent/blob/master/cmd/system-probe/modules/tcp_queue_tracer.go#L18